spacy/pipeline/tok2vec.py

```diff
@@ -291,7 +291,7 @@ def forward(model: Tok2VecListener, inputs, is_train: bool):
         # of data.
         # When the components batch differently, we don't receive a matching
         # prediction from the upstream, so we can't predict.
-        if not all(doc.tensor.size for doc in inputs):
+        if not any(doc.tensor.size for doc in inputs):
```

Comment:
Thanks for finding this!

I can't decide whether it's more or less brittle to do it as a loop?

```
outputs = []
width = model.get_dim("nO")
for doc in docs:
    if doc.tensor.size == 0:
        outputs.append(model.ops.alloc2f(len(doc), width))
    else:
        outputs.append(doc.tensor)
```

Maybe the loop is easier to understand at least?

Reply:
I'm definitely fine with the loop, but I still think we should consider warning here. In most cases, this code path results in trouble. I know we need it for initialization & shape inference, but we could have a clever way (hopefully) to avoid the warnings in that specific case, no?

Reply:
I see what you're saying but I don't want to add more guessing. Let's get the loop in and get this fix deployed, and then plot our next mitigation. 😬